mayastor/src/lvs/pool.rs

```diff
@@ -0,0 +1,511 @@
+        };
+
+        r.await
+            .expect("callback gone while exporting lvs")
+            .to_result(|e| Error::Export {
+                err: Errno::from_i32(e),
+                msg: format!("failed to export pool {}", pool),
+            })?;
+
+        info!("pool {} exported successfully", pool);
+        bdev_destroy(&base_bdev.bdev_uri().unwrap())
+            .await
+            .map_err(|e| Error::Destroy {
+                source: e,
+                msg: format!(
+                    "pool {} destroyed but failed to delete base_bdev {}",
+                    pool,
+                    base_bdev.name()
+                ),
+            })?;
+        Ok(())
+    }
+
+    /// unshare all lvols prior to export or destroy
+    async fn unshare_all(&self) {
+        for l in self.lvols().unwrap() {
+            // notice we dont use the unshare impl of the bdev
+            // here. we do this to avoid the on disk persistence
+            let bdev = l.as_bdev();
+            if let Err(e) = bdev.unshare().await {
+                error!("failed to unshare lvol {} error {}", l, e.to_string())
+            }
+        }
+    }
+
+    /// share all lvols who have the shared property set, this is implicitly
+    /// shared over nvmf
+    async fn share_all(&self) {
+        if let Some(lvols) = self.lvols() {
+            for l in lvols {
+                if let Ok(prop) = l.get(PropName::Shared).await {
+                    match prop {
+                        PropValue::Shared(true) => {
+                            if let Err(e) = l.share_nvmf().await {
+                                error!(
+                                    "failed to share {} {}",
+                                    l.name(),
+                                    e.to_string()
+                                );
+                            }
+                        }
+                        PropValue::Shared(false) => {
+                            debug!("{} not shared on disk", l.name())
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /// destroys the given pool deleting the on disk super blob before doing so,
+    /// un share all targets
+    #[allow(clippy::unit_arg)]
+    #[instrument(level = "debug", err)]
+    pub async fn destroy(self) -> Result<(), Error> {
+        let pool = self.name().to_string();
+        let (s, r) = pair::<i32>();
+
+        // when destroying a pool unshare all volumes
+        self.unshare_all().await;
+
+        let base_bdev = self.base_bdev();
+
+        unsafe {
+            vbdev_lvs_destruct(
+                self.0.as_ptr(),
+                Some(Self::lvs_op_cb),
+                cb_arg(s),
+            )
+        };
+
+        r.await
+            .expect("callback gone while destroying lvs")
+            .to_result(|e| Error::Export {
+                err: Errno::from_i32(e),
+                msg: format!("failed to export pool {}", pool),
```

Comment:
should this be "failed to destroy the pool {}"?

Reply:
Yes, I will fix that later, thanks for spotting.